### PR TITLE
Remove ridiculous macOS OpenGL deprecation warnings

### PR DESCRIPTION
**Pull request by milesrout**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -7,4 +7,4 @@
 	branch = releases/tag/v0.5.2-fix
 [submodule "deps/selene"]
 	path = deps/selene
-url=https://github.com/adam4813/Selene
\ No newline at end of file
+	url = https://github.com/adam4813/selene
```
Comment (adam4813) [inline .gitmodules]: This will most likely merge conflict as I changed the source back to the upstream repo

Comment (milesrout) [reply .gitmodules]: Ah whoops!

File: client/CMakeLists.txt [hunk 2]
```diff
@@ -22,6 +22,12 @@ else ()
 	set(trillek-client_PROGRAM "trillek-client")
 endif ()
 
+# Rather wonderfully, Apple decided to mark every single fucking OpenGL function
+# as deprecated because they have 'deprecated' OpenGL. Brilliant!
+if(APPLE)
+    add_definitions(-DGL_SILENCE_DEPRECATION)
```
Comment (adam4813) [inline client/CMakeLists.txt]: Nice, good find!

**Comment by milesrout** [comment]
Builds on my computer

**Pull request merged**